### PR TITLE
don't use zlib-ng's -Wl,--version-script in tests

### PR DESCRIPTION
**Pull request by haampie**
lld 18 errors when a version script assigns a version to a symbol that
is not defined in the object files. Therefore configure scripts should
not use the -Wl,--version-script,zlib.map script -- all tests that create a
dynamic library will fail

### CODE REVIEW

File: configure [hunk 1]
```diff
@@ -630,6 +630,29 @@ fi
 
 echo >> configure.log
 
+# check for version script support
+cat > $test.c <<EOF
+int foo() { return 0; }
```
Comment (haampie) [inline configure]:
```suggestion
int foo(void) { return 0; }
```

**Review by Dead2 — APPROVED** [review]
LGTM

**Comment by haampie** [comment]
@Dead2 do you do backports for 2.1.x?

**Comment by Dead2** [comment]
@haampie Yes, I was thinking of backporting this to the `2.1.x` branch.

**Pull request merged**